module/VuFind/src/VuFind/Db/Entity/Search.php

```diff
@@ -302,7 +315,15 @@ public function setSaved(bool $saved): static
      */
     public function getSearchObject(): ?\VuFind\Search\Minified
     {
-        return $this->searchObject ? unserialize($this->searchObject) : null;
+        if (!$this->searchObject) {
+            return null;
+        }
+        // Normalize resource
+        if (is_resource($this->searchObject)) {
```

Comment:
I think we might be able to refactor this resource resolution logic to a [postLoad event hook](https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/events.html#reference-events-post-load), which might be a more elegant approach.